recipes/license_server.rb

```diff
@@ -70,6 +70,21 @@
   notifies :touch, 'file[splunk-marker]', :immediately
 end
 
+ruby_block 'check license overuse' do
+  block do
+    # Find the total available license quota
+    total_available_license_quota = 0
+    existing_license = Dir.glob("#{node['splunk']['home']}/etc/licenses/enterprise/*.lic*")
+    existing_license.each do |file|
+      license_file = Nokogiri::XML(File.open(file))
+      max_quota = license_file.at_xpath('/license/payload/expiration_time/text()').to_s.to_i > Time.now.to_i ? license_file.at_xpath('/license/payload/quota/text()').to_s.to_i : 0
+      total_available_license_quota += max_quota
+    end
+    total_allotted_pool_size = node.run_state['total_allotted_pool_size']
```

Comment:
This could be nil if someone is not specifying pools, in which case we wouldn't want to check this?